go/services/reports/pkg/openai/batch_poller.go

```diff
@@ -0,0 +1,219 @@
+package openai
+
+import (
+	"fmt"
+	"github.com/Magpie-Monitor/magpie-monitor/pkg/envs"
+	"go.uber.org/zap"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+)
+
+const (
+	OpenAiBatchStatus__Validating = "validating"
+	OpenAiBatchStatus__Failed     = "failed"
+	OpenAiBatchStatus__InProgress = "in_progress"
+	OpenAiBatchStatus__Finalizing = "finalizing"
+	OpenAiBatchStatus__Completed  = "completed"
+	OpenAiBatchStatus__Expired    = "expired"
+	OpenAiBatchStatus__Cancelling = "cancelling"
+	OpenAiBatchStatus__Cancelled  = "cancelled"
+)
+
+const BATCH_AWAITING_INTERVAL_SECONDS = 10
+
+type BatchPoller struct {
+	batches                chan *Batch
+	client                 *Client
+	pendingBatchRepository PendingBatchRepository
+	pollingIntervalSeconds int
+}
+
+func NewBatchPoller(client *Client, pendingBatchRepository PendingBatchRepository) *BatchPoller {
+
+	envs.ValidateEnvs("Missing envs for openai batch poller", []string{POLLING_INTERVAL_SECONDS_KEY})
+
+	pollingIntervalSeconds := os.Getenv(POLLING_INTERVAL_SECONDS_KEY)
+	pollingIntervalSecondsInt, err := strconv.Atoi(pollingIntervalSeconds)
+
+	if err != nil {
+		panic(fmt.Sprintf("%s is not a number", POLLING_INTERVAL_SECONDS_KEY))
+	}
+
+	poller := &BatchPoller{
+		batches:                make(chan *Batch),
+		client:                 client,
+		pendingBatchRepository: pendingBatchRepository,
+		pollingIntervalSeconds: pollingIntervalSecondsInt,
+	}
+
+	go poller.Start()
```

Comment:
Similar to my previous comment, starting poller in the constructor seems quite unexpected